### PR TITLE
fix: check column references in ORDER BY (#1411)

### PR DESCRIPTION
**Pull request by akutschera**
MySQL and Sqlite store the information for the order by columns in a WindowClause while PostgreSQL stores it in a SortClause.
That's why I added two new if-statements.

Refs: #1411

### CODE REVIEW

File: internal/compiler/parse_test.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
+package compiler
```
Comment (kyleconroy) [inline internal/compiler/parse_test.go:1]: I take a bit of a different approach to testing sqlc, opting mainly for end-to-end tests that don't verify internal behavior. Can you take these tests and move them into internal/endtoend/testdata/order_by_non_existing_column?

**Review by kyleconroy — CHANGES_REQUESTED** [review]
Adding additional validation like this is great, but it will cause queries that used to work to fail. If the validation code is wrong, users won't be able to get their queries working. 

Let's add a new configuration option [here](https://github.com/kyleconroy/sqlc/blob/main/internal/config/config.go#L94) called `validate_order_by` that defaults to true (make it a pointer to a bool). Then we can update the error message telling users they can turn it off by setting `validate_order_by: false`.

File: internal/compiler/parse_test.go (hunk 1 above)
Comment (akutschera) [reply internal/compiler/parse_test.go:1]: That I can do. We actually only need these tests for order by. The group by tests are in internal/endtoend/testdata/invalid_group_by_reference already

**Comment by kyleconroy** [comment]
sqlc getting more strict over time is a good thing. `strict_function_checks` works in V1 as well, so we can put `validate_order_by` in the same spot.

**Comment by akutschera** [comment]
I added the ```validate_order_by``` config option. It works even though it feels a bit awkward to pass this value through multiple function calls. I am open to suggestions to improve this before it gets merged.
I will do the documentation in a separate PR because I also want to add documentation for ```strict_function_checks``` in V1 and I don't think this belongs to this PR.

**Review by kyleconroy — APPROVED** [review]

**Comment by Stumble** [comment]
Hey @akutschera ,

There might be a minor bug in checking ambiguous references when joning tables while using alias, see https://github.com/kyleconroy/sqlc/issues/2398

**Pull request merged**